### PR TITLE
rewrite the library: simpler design and implementation, unit tests

### PR DESCRIPTION
**Pull request by amtoine**
ok, this is quite a big one but bear with me, i'll explain everything :wink: 

## changelog
- remove `docs/` => out of date
- remove `toolkit/` => out of date
- remove `sugar completions git` => let's use `carapace` or use the ones from `nu_scripts`
- remove `sugar dotfiles` => a bit out of scope
- i left the rest of the `sugar` module but moved it to `nu-git-manager/sugar/` => i'd like to refactor the namespace of this as well in a follow-up PR

> **Note**  
> i've written command names, arguments, type annotations and tests for internal tool commands with the goal of being self-contained in mind without requiring lengthy documentation :crossed_fingers:

- the `nu-git-manager git` **private** module is about Git related tool commands
    - `git url` focuses on manipulating Git URLs
- the `nu-git-manager fs` **private** module is about interacting with the filesystem
    - `fs store` focuses on interacting with the local store of repositories
- tests cover `git url parse-git-url` and `git url get-fetch-push-urls` with `nupm test`
- not sure how to properly test `fs store` without interacting with the host environment :thinking:
- the `git` and `fs` modules are basically simplified and encapsulated versions of the previous `utils utils` module
- `gm` itself
    - i've removed the `ghq` API => it was making the code too complex and honestly not sure we care about that kind of compatibility
    - not that `gm` is not a module anymore, but a *true* command with subcommands, i.e. you don't `use nu-git-manager gm` but `use nu-git-manager *` or `use nu-git-manager ["gm list", "gm clone"]`
    - i think the easiest here is to just read the docstrings and the arguments of the `export`ed commands
    - now one of the biggest changes, and the one that gave me the motivation to do the rewrite, was that, when cloning a repo with the SSH protocol, you end up with the `origin` remote setup with SSH for both FETCH and PUSH. with that, even if the repo is public and you only `git fetch`, you still have to write the passphrase for the SSH key, which is a bit of a pain... :laughing: now, as you can see in the examples of `gm clone`, one can use the `--ssh`, `--fetch` and `--push` options to tweak that in the *clone* step :ok_hand:
    there's even autocompletion for `--fetch` and `--push` :partying_face:
- the `README` has been updated => [have a read](https://github.com/amtoine/nu-git-manager/tree/refactor#nu-git-manager)
- finally i bump the version to `0.2.0`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,46 +1,64 @@
...
 - `gh` (optional) 2.29.0 (used by `sugar gh`)
+    - with Pacman and `pacman -S community/github-cli`
+    - with Nix and `nix run nixpkgs#gh`
+- `find` 4.9.0
+    - with Pacman and `pacman -S core/findutils`
+    - with Nix and `nix run nixpkgs#findutils`
 
-## :recycle: installation [here](docs/installation/)
+## :recycle: installation [[toc](#table-of-content)]
+- install [Nupm] (**recommended**) by following the [Nupm instructions]
```
Comment (melMass) [inline README.md:34]: Maybe add a small section about how to do without, like add the nu-git-manager root folder to modules path

**Comment by melMass** [comment]
I just learned that Github support nushell syntax!! 😮 
```nushell
manage your Git repositories with the main command of `nu-git-manager`

Usage:
  > gm

Subcommands:
  gm clone - clone a remote Git repository into your local store
  gm list - list all the local repositories in your local store
  gm remove - remove one of the repositories from your local store
  gm root - get the root of the local store of repositories managed by `nu-git-manager`

Flags:
  -h, --help - Display the help message for this command

Input/output types:
  ╭───┬─────────┬─────────╮
  │ # │  input  │ output  │
  ├───┼─────────┼─────────┤
  │ 0 │ nothing │ nothing │
  ╰───┴─────────┴─────────╯

```

**Comment by melMass** [comment]
Great work, I agree with all the directions on simplification.
It all looks good to me, I'm just going to fully test it on windows and mac as I already see a few non cross platform commands.
I asked on discord but asking here too just in case: If I want to add some shared utils by the modules, not exposed by the lib what would you recommend?

**Comment by amtoine** [comment]
> I just learned that Github support nushell syntax!! 😮
> 
pretty cool huh :wink:

**Comment by amtoine** [comment]
let's go with this :partying_face: 

thanks for the review @melMass :pray:

**Pull request merged**